dlp/snippets/risk/risk_test.go

```diff
@@ -152,3 +158,150 @@ func cleanupPubsub(t *testing.T, client *pubsub.Client, topicName, subName strin
 		}
 	}
 }
+
+func TestMain(m *testing.M) {
+	tc := testutil.Context{}
+	tc.ProjectID = os.Getenv("GOLANG_SAMPLES_PROJECT_ID")
+	if tc.ProjectID == "" {
+		tc.ProjectID = os.Getenv("")
+	}
+
+	createBigQueryDataSetId(tc.ProjectID)
+	createTableInsideDataset(tc.ProjectID, dataSetID)
+	m.Run()
+	deleteBigQueryAssets(tc.ProjectID)
+}
+
+func TestCalculateKAnonymityWithEntityId(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	buf := new(bytes.Buffer)
+	err := calculateKAnonymityWithEntityId(buf, tc.ProjectID, dataSetID, tableID, "title", "contributor_ip")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if got, want := buf.String(), "Created job"; !strings.Contains(got, want) {
+		t.Errorf("CalculateKAnonymityWithEntityId got %s, want substring %q", got, want)
+	}
+	if got, want := buf.String(), "Quasi-ID values"; !strings.Contains(got, want) {
+		t.Errorf("CalculateKAnonymityWithEntityId got %s, want substring %q", got, want)
+	}
+	if got, want := buf.String(), "Job status: DONE"; !strings.Contains(got, want) {
+		t.Errorf("CalculateKAnonymityWithEntityId got %s, want substring %q", got, want)
+	}
+	if got, want := buf.String(), "Bucket size range:"; !strings.Contains(got, want) {
+		t.Errorf("CalculateKAnonymityWithEntityId got %s, want substring %q", got, want)
+	}
+	if got, want := buf.String(), "Class size: 1"; !strings.Contains(got, want) {
+		t.Errorf("CalculateKAnonymityWithEntityId got %s, want substring %q", got, want)
+	}
+	if got, want := buf.String(), "Job name:"; !strings.Contains(got, want) {
+		t.Errorf("CalculateKAnonymityWithEntityId got %s, want substring %q", got, want)
+	}
+
+}
+
+func createBigQueryDataSetId(projectID string) error {
+
+	ctx := context.Background()
+
+	client, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	meta := &bigquery.DatasetMetadata{
+		Location: "US", // See https://cloud.google.com/bigquery/docs/locations
+	}
+
+	if err := client.Dataset(dataSetID).Create(ctx, meta); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createTableInsideDataset(projectID, dataSetID string) error {
+	ctx := context.Background()
+	client, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	sampleSchema := bigquery.Schema{
+		{Name: "user_id", Type: bigquery.StringFieldType},
+		{Name: "age", Type: bigquery.IntegerFieldType},
+		{Name: "title", Type: bigquery.StringFieldType},
+		{Name: "score", Type: bigquery.StringFieldType},
+	}
+
+	metaData := &bigquery.TableMetadata{
+		Schema:         sampleSchema,
+		ExpirationTime: time.Now().AddDate(1, 0, 0), // Table will be automatically deleted in 1 year.
+	}
+
+	tableRef := client.Dataset(dataSetID).Table(tableID)
```

Comment:
issue: we should probably add UUIDs to the `tableID` to avoid resource contention.

Reply:
updated here : https://github.com/GoogleCloudPlatform/golang-samples/blob/5abe0a353dba6772a246ff7a863667e9a6cab21d/dlp/snippets/risk/risk_test.go#L162